src/city/health.c

```diff
@@ -396,31 +396,23 @@ int city_health_get_global_sickness_level(void)
         }
     }
 
-    int plague_incoming;
-
-    if (max_sickness_level >= HIGH_SICKNESS_LEVEL) {
-        plague_incoming = 2;
-    } else if (max_sickness_level >= MEDIUM_SICKNESS_LEVEL) {
-        plague_incoming = 1;
-    } else {
-        plague_incoming = 0;
-    }
-
     if (building_number == 0) {
         return SICKNESS_LEVEL_LOW;
     }
 
     int global_rating = building_sickness_level / building_number;
```

Comment:
From what I understand of the code, I think this whole thing can be changed to:

```
    if (max_sickness_level >= MAX_SICKNESS_LEVEL) { // one or many houses is plagued
        return SICKNESS_LEVEL_PLAGUE;
    }
    if (max_sickness_level >= HIGH_SICKNESS_LEVEL) { // one or many houses have sickness_level >= 90
        return SICKNESS_LEVEL_HIGH;
    }
    if (max_sickness_level >= MEDIUM_SICKNESS_LEVEL) { // one or many houses have sickness_level >= 60
        return SICKNESS_LEVEL_MEDIUM;
    }
    return SICKNESS_LEVEL_LOW;
```

This is because, from what I get from the code, the max sickness level always overrides the global sickness level, so there are no cases where the global sickness level is considered.

This is because with the global being an average, it will always be lower or equal to the max sickness level.

Reply:
I think if global rating is ~50 with max level under 60, we want SICKNESS_LEVEL_MEDIUM not LOW

Reply:
Oh I see. You're right but the code is unclear. I mean I missed that cased and I looked at those lines for something like 5 minutes.

I think a weighed calculation would be clearer then, like:

```
int current_sickness = (calc_adjust_with_percentage(global_sickness, 80) + calc_adjust_with_percentage(max_sickness, 20)) / 2;
```

The ratio of 80 global / 20 max could be changed obviously, to see what would work better.

Then using the `if`s above with the `current_sickness` variable.

Reply:
So the new code would be (before I push something wrong ^^)

```
if (building_number == 0) {
        return SICKNESS_LEVEL_LOW;
    }

    int current_sickness = (calc_adjust_with_percentage(global_sickness, 80) + calc_adjust_with_percentage(max_sickness, 20)) / 2;

    if (current_sickness >= MAX_SICKNESS_LEVEL) { // one or many houses is plagued
        return SICKNESS_LEVEL_PLAGUE;
    }
    if (current_sickness >= HIGH_SICKNESS_LEVEL) { // one or many houses have sickness_level >= 90
        return SICKNESS_LEVEL_HIGH;
    }
    if (current_sickness >= MEDIUM_SICKNESS_LEVEL) { // one or many houses have sickness_level >= 60
        return SICKNESS_LEVEL_MEDIUM;
    }
    
    return SICKNESS_LEVEL_LOW;
```

`int global_rating` can be removed? It is not used elsewhere, it seems?

Reply:
Yes that's it. You can technically remove the `global_sickness_level` variable, but maybe that line will get too crowded with calculations if you do. Best keep it to keep things clear.

@dvincent56 What do you think about this alternative?

Reply:
Seems good. 
tbh, I was confused when I tried to get the real sickness rating.
The problem was, taking into account the average level but also each house maximum level to determine this "rating".
@Ouaz The advisor display looks correct like this ?

Reply:
I didn't test it yet. I have to push the changes so Github can build the exe (I don't know doing it locally ^^)

Reply:
Let me know if you want to learn how to do it locally, it's much more convenient!

Reply:
Oh oh, it seems I finally did something wrong:

```
D:\a\augustus\augustus\src\city\health.c: In function 'city_health_get_global_sickness_level':
D:\a\augustus\augustus\src\city\health.c:404:57: error: 'global_sickness' undeclared (first use in this function); did you mean 'global_sickness_level'?
  404 |     int current_sickness = (calc_adjust_with_percentage(global_sickness, 80) + calc_adjust_with_percentage(max_sickness, 20)) / 2;
      |                                                         ^~~~~~~~~~~~~~~
      |                                                         global_sickness_level
D:\a\augustus\augustus\src\city\health.c:404:57: note: each undeclared identifier is reported only once for each function it appears in
D:\a\augustus\augustus\src\city\health.c:404:108: error: 'max_sickness' undeclared (first use in this function); did you mean 'max_sickness_level'?
  404 |     int current_sickness = (calc_adjust_with_percentage(global_sickness, 80) + calc_adjust_with_percentage(max_sickness, 20)) / 2;
      |                                                                                                            ^~~~~~~~~~~~
      |                                                                                                            max_sickness_level
D:\a\augustus\augustus\src\city\health.c:403:9: warning: unused variable 'global_sickness_level' [-Wunused-variable]
  403 |     int global_sickness_level = SICKNESS_LEVEL_LOW;
```